### DISCUSSION TITLE
Multiple static vectors

### DISCUSSION THREAD

**Discussion by thejamesmarq** [explosion/spaCy#8179] [Help: Coding & Implementations] [answered]
In some other projects I'm working on I've had some success using multiple embeddings using different static vectors (e.g. FastText and some custom built Word2Vec), the outputs of the embeddings then being concatenated.

In Spacy I've been able to use static vectors, but it looks like there's only a way to use one set of vectors. Is there a way to create multiple embeddings using multiple vectors that can be concatenated?

If this is only possible using code, that's fine, but I'm curious mostly if it can be done using the configs, since that is one of the big selling points for me in Spacy.

**Comment by polm** [comment, marked as answer]
There is no special functionality for this, no. 

You should be able to pre-concatenate your vectors and load them in just like normal vectors though.

**Reply by thejamesmarq** [reply]
Gotcha. I was imagining something more like having multiple `tok2vec` components for a model, each creating an embedding using different static vectors (possibly using a CNN as part of it).

I'll take a look at whether I get the same performance doing static vector concatenation as a preprocessing step though.

**Reply by polm** [reply]
Good luck! This paper on alternatives to simple concatenation might also be interesting:

https://arxiv.org/abs/1805.09843
